tre-command: init at 0.2.2

dduan opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change
tre is a improved version of the command `tree`. It's main additions:

1. colored output
2. ignores paths specified in .gitignore
3. editor alias for each entity listed

Source/Homepage: https://github.com/dduan/tre

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

I'd like to be maintainer of this package for Nix. (I'm the author of the CLI tool).

mmahut commented on maintainers/maintainer-list.nix:
```diff
@@ -1563,6 +1563,12 @@
     githubId = 14032;
     name = "Daniel Brockman";
   };
+  dduan = {
```
Please include this a separate commit.

mmahut commented on pkgs/tools/system/tre/default.nix:
```diff
@@ -0,0 +1,24 @@
+with import <nixpkgs> {};
+
+rustPlatform.buildRustPackage rec {
+  name = "tre-${version}";
```
Please use ```pname```.

mmahut commented on pkgs/tools/system/tre/default.nix:
```diff
@@ -0,0 +1,24 @@
+with import <nixpkgs> {};
```
Please don't import all the packages.

mmahut commented on pkgs/tools/system/tre/default.nix:
```diff
@@ -0,0 +1,24 @@
+
+  meta = with stdenv.lib; {
+    description = "Tree command, improved";
+    homepage = https://github.com/dduan/tre;
```
Please quote the url.

dduan committed:
Add dduan as a maintainer

dduan committed:
tre: init at 0.2.2

tre is a improved version of the command `tree`. Its main additions:

1. colored output
2. ignores paths specified in .gitignore
3. editor alias for each entity listed

Source/Homepage: https://github.com/dduan/tre

dduan commented:
@mmahut thanks for your review! I've addressed your feedbacks.

mmahut commented on pkgs/tools/system/tre/default.nix:
```diff
@@ -0,0 +1,24 @@
+{ rustPlatform, fetchFromGitHub, stdenv }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "tre-${version}";
```
```suggestion
  pname = "tre";
```

The name is contrusted of ```${pname}-${version}``` so you should use just the name itself.

mmahut commented on pkgs/tools/system/tre/default.nix:
```diff
@@ -0,0 +1,24 @@
+
+rustPlatform.buildRustPackage rec {
+  pname = "tre-${version}";
+  version = "v0.2.2";
```
```suggestion
  version = "0.2.2";
```

Please move the ```v``` in the tag to rev, as noted below.

mmahut commented on pkgs/tools/system/tre/default.nix:
```diff
@@ -0,0 +1,24 @@
+  src = fetchFromGitHub {
+    owner = "dduan";
+    repo = "tre";
+    rev = "${version}";
```
```suggestion
    rev = "v${version}";
```

mmahut commented on pkgs/top-level/all-packages.nix:
```diff
@@ -21304,6 +21304,8 @@ in
 
   tdrop = callPackage ../applications/misc/tdrop { };
 
+  tre-command = callPackage ../tools/system/tre {};
```
Any reason for renaming this?

dduan replied:
yes, there's a existing library named `tre` here already.

mmahut replied:
Ah, I see, can we rename the directory too in that case (pname can stay as it is)?

dduan replied:
done

mmahut replied:
I think you need to change the path here too.

dduan replied:
Yup done

mmahut commented:
@GrahamcOfBorg build  tre-command

dduan commented:
@mmahut lemme know if there's further improvement to be made in order for this to be merged. Thanks for your help so far!

mmahut commented:
@GrahamcOfBorg build tre-command

mmahut commented:
Looks good to me. Thank you Daniel for your contribution and welcome! :)
